scripts/compress_images.py

```diff
@@ -0,0 +1,233 @@
+# coding: utf-8
+#
+# Copyright 2025 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Script to compress images in the repository using GraphicsMagick."""
+
+from __future__ import annotations
+
+import logging
+import os
+import pathlib
+import shutil
+import subprocess
+import sys
+import tempfile
+
+from PIL import Image
+from typing import List, TypedDict, Union
+
+
+class CompressedImageInfo(TypedDict):
+    """Type definition for compressed image information."""
+
+    path: pathlib.Path
+    original_size: int
+    new_size: int
+
+
+class ImageCompressor:
+    """A class to handle image compression using GraphicsMagick."""
+
+    IMAGE_EXTENSIONS_SUPPORTING_ZIP_COMPRESSION = {'.png'}
+    IMAGE_EXTENSIONS_SUPPORTING_LZW_COMPRESSION = {'.jpg', '.jpeg', '.webp'}
+    ALL_IMAGES_EXTENSIONS = (
+        IMAGE_EXTENSIONS_SUPPORTING_ZIP_COMPRESSION |
+        IMAGE_EXTENSIONS_SUPPORTING_LZW_COMPRESSION
+    )
+    TOLERANCE = 0.99
+    OUTPUT_DIR = 'compressed_images'
+
+    def __init__(
+        self,
+        input_path: Union[str, pathlib.Path],
+        output_dir: str = OUTPUT_DIR,
+        tolerance: float = TOLERANCE
+    ):
+        """Initialize the ImageCompressor.
+
+        Args:
+            input_path: str. Images directory.
+            output_dir: str. Directory to store compressed images.
+            tolerance: float. Compression threshold.
+        """
+        self.input_path = pathlib.Path(input_path)
+        self.output_dir = output_dir
+        self.tolerance = tolerance
+        self.logger = logging.getLogger(__name__)
+
+    def get_compression_type(self, file_extension: str) -> str:
+        """Determine compression type based on file extension.
+
+        Args:
+            file_extension: str. The file extension of the image.
+
+        Returns:
+            type: str. Compression type (Zip or LZW).
+        """
+        return (
+            'Zip' if file_extension in 
+            self.IMAGE_EXTENSIONS_SUPPORTING_ZIP_COMPRESSION
+            else 'LZW'
+        )
+
+    def find_compressible_images(self) -> List[CompressedImageInfo]:
+        """Find images that can be compressed further.
+
+        Returns:
+            result_images: List[]. List of compressible images.
```

Comment:
```suggestion
            result_images: list(CompressedImageInfo). List of compressible images.
```

We use a different style for our docstring types. Here's a good example you can reference: https://github.com/oppia/oppia/blob/8b3b4cec44cd328ca92e02c106f8bdc3172f6bbe/scripts/servers.py#L54